### PR TITLE
darwin: implement node:os.freemem()

### PR DESCRIPTION
**Pull request by nektro**
Closes https://github.com/oven-sh/bun/issues/3944, linux and windows are already implemented

```
❯ node index.js
Free memory: 1839923200
Free memory: 1754MB
```

```
❯ bun-debug index.js
Free memory: 1633452032
Free memory: 1557MB
```

### CODE REVIEW

File: src/bun.js/bindings/OsBinding.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include <mach/mach_host.h>
+#include <mach/mach_init.h>
+#include <mach/message.h>
+#include <mach/vm_statistics.h>
+#include <unistd.h>
+
+// Adapted from libuv darwin uv_get_free_memory, MIT
+extern "C" uint64_t Bun__Os__getFreeMemory(void)
+{
+    vm_statistics_data_t info;
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/OsBinding.cpp:14]: We don't need to worry about zero-initializing this?

**Comment by Jarred-Sumner** [comment]
can you add a test that checks the number is != `1024 * 1024` and != 0?

**Comment by robobun** [comment]
@nektro, your commit 83c549da65904dc875aa1d742b40a21d67cb85f7 is building: [`#1075`](https://buildkite.com/bun/bun/builds/1075)

File: src/bun.js/bindings/OsBinding.cpp (hunk 1 above)
Comment (nektro) [reply src/bun.js/bindings/OsBinding.cpp:14]: no, it's initialized by `host_statistics()`

**Pull request merged**